### DISCUSSION TITLE
Checking the build output when assessing the applicability of cache

### DISCUSSION THREAD

**Discussion by dtinth** [ElMassimo/vite_ruby#262] [Q&A] [answered]
Hi! First of all, thanks for this great library, and especially thanks for making it usable with Rails 4. You have made the lives of people who maintain Ruby on Rails project much more pleasant!

We’re running into a problem where a deployment would fail at runtime due to missing Vite assets.

> ViteRuby::MissingEntrypointError
> Vite Ruby can't find entrypoints/application.ts in public/vite/manifest.json or public/vite/manifest-assets.json.

Related discussions:

- https://github.com/ElMassimo/vite_ruby/discussions/186
- https://github.com/ElMassimo/vite_ruby/discussions/197

We’re not using Capistrano to deploy our app; we use @cloud66. Upon further digging, we found that when deploying, **the cache folder is shared between deployments.** During deployment, Cloud66 performs this step:

```
* executing `deploy:cache_symlink'
* executing "ln -s /var/deploy/app/web_head/shared/cache /var/deploy/app/web_head/releases/20220919012345/tmp/cache"
```

(It seems that Cloud66 uses Capistrano under-the-hood, however we don’t have a Capfile or any mention of Capistrano in our source code.)

Meanwhile, the `public/vite` folder is specific to each build. This meant that when we deploy an app without a frontend code change, this happens:

1. Vite Ruby sees `tmp/cache/vite/last-build-*.json` from the previous build and skips building.
2. The folder `public/vite` did not get created at all.

Since we don’t want to add Capistrano to our project yet (we plan to migrate to a Dockerized build), what we do is setting `VITE_RUBY_BUILD_CACHE_DIR` to `tmp/vite/cache` instead of `tmp/cache/vite` (so they are not shared).

<img width="999" alt="image" src="https://user-images.githubusercontent.com/193136/191170885-90802076-8c20-4985-a323-dc1565906ac9.png">

To make the process safer, maybe Vite Ruby can also check if the manifest file exists. That is, if the output manifest file (e.g. `public/vite/manifest.json` or `public/vite/manifest-assets.json`) doesn’t exist, maybe the cache file should be ignored.

**Comment by ElMassimo** [comment]
Hi Thai!

Ignoring the cache when the manifest is missing would prevent the error, sounds like a good idea!

On the other hand, it will make it harder to realize that Vite Ruby is not reusing previously built assets.

Perhaps a middle ground would be to log a warning then the cache key matches, but the manifest is missing.

Let me think about it, thanks for bringing this up 😃

**Comment by ElMassimo** [comment, marked as answer]
Implemented in:

- https://github.com/ElMassimo/vite_ruby/pull/275

Thanks for the suggestion Thai! 😃

**Reply by dtinth** [reply]
Thanks! 😄
